src/runtime/hexagon_remote/halide_host.cpp

```diff
@@ -116,19 +117,18 @@ __attribute__((weak)) void remote_register_buf(void* buf, int size, int fd);
 
 void halide_hexagon_host_malloc_init() {
     pthread_mutex_init(&allocations_mutex, NULL);
+    ion_fd = open("/dev/ion", O_RDONLY, 0);
```

Comment:
The code as it stands seems like it will leak fds if you have a nested
init/release pair.

On Thu, Aug 18, 2016 at 10:25 AM, Dillon Sharlet <notifications@github.com>
wrote:

> In src/runtime/hexagon_remote/halide_host.cpp
> <https://github.com/halide/Halide/pull/1452#discussion_r75350848>:
>
> > @@ -116,19 +117,18 @@ __attribute__((weak)) void remote_register_buf(void* buf, int size, int fd);
> >
> >  void halide_hexagon_host_malloc_init() {
> >      pthread_mutex_init(&allocations_mutex, NULL);
> > +    ion_fd = open("/dev/ion", O_RDONLY, 0);
>
> open/close should be ref counted, so there's no problem running this
> multiple times, as long as each open call has a matching close.
>
> That said, the rest of this init/deinit code is not thread safe. It's not
> urgent to fix this because we know how these are called, but we should fix
> this at some point.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/halide/Halide/pull/1452/files/3226aa391ae3e77655426ef61e91f313aaa6f8bb#r75350848>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAfdRhR4y49Z3n5KAf-7-mhqlb1IRFjRks5qhJVxgaJpZM4Jl-at>
> .
>